WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostMoreButtonUiStateBuilder.kt

```diff
@@ -42,136 +43,144 @@ class ReaderPostMoreButtonUiStateBuilder @Inject constructor(
     fun buildMoreMenuItemsBlocking(
         post: ReaderPost,
         onButtonClicked: (Long, Long, ReaderPostCardActionType) -> Unit
-    ): MutableList<SecondaryAction> {
-        val menuItems = mutableListOf<SecondaryAction>()
+    ): MutableList<ReaderPostCardAction> {
```

Comment:
1. Praise (❤️): Kudos for picking up on the Detekt's suggestion and splitting up this function into multiple ones!
1. Suggestion (💡): Consider extracting to more functions out of this parent function, even for those with additional logic. For example, see the below 3 additional extracted `checkAndAddMenuItemForXyz(...)` functions:
```
    fun buildMoreMenuItemsBlocking(
        post: ReaderPost,
        onButtonClicked: (Long, Long, ReaderPostCardActionType) -> Unit
    ): MutableList<ReaderPostCardAction> {
        val menuItems = mutableListOf<ReaderPostCardAction>()
        val isPostFollowed = readerPostTableWrapper.isPostFollowed(post)
        menuItems.add(buildVisitSite(onButtonClicked))
        checkAndAddMenuItemForSiteNotifications(menuItems, isPostFollowed, post, onButtonClicked)
        checkAndAddMenuItemForPostSeenUnseen(menuItems, post, onButtonClicked)
        menuItems.add(buildShare(onButtonClicked))
        menuItems.add(buildFollow(isPostFollowed, onButtonClicked))
        menuItems.add(SpacerNoAction())
        checkAndAddMenuItemForBlockSite(menuItems, isPostFollowed, onButtonClicked)
        menuItems.add(buildReportPost(onButtonClicked))
        return menuItems
    }
```